core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -1489,7 +1489,7 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
   private[spark] def killAndReplaceExecutor(executorId: String): Boolean = {
     schedulerBackend match {
       case b: CoarseGrainedSchedulerBackend =>
-        b.killExecutors(Seq(executorId), replace = true)
+        b.killExecutors(Seq(executorId), replace = true, force = false)
```

Comment:
yes. If it for dead executor, we should force kill that and replace a new one. No matter the executor is busy or not.